### PR TITLE
Issue #223: Update samples within doc to reflect latest code changes

### PR DESCRIPTION
**Pull request by danilko**
…amples within doc to reflect latest code changes

*Issue #, if available:*
https://github.com/aws-quickstart/ssp-amazon-eks/issues/223: Update samples within doc to reflect latest code changes

*Description of changes:*

Cover most ones in the above issue. The CDK version is already updated (but may just not deploy to site?)

Also update `README.md` and quickstart guide sample to be same for easier future maintenance. 

One note about `README` , the original sample code did not work by default as `team` and `scope` is undefined. Originally discussion with @askulkarni2 is to place a dummy sample. But after thinking about it, by placing a dummy team sample will create a default account allow to access the cluster and increase security attack vector, so maybe better to not create by default. So just remove `scope` and `team` from default sample. Is certainly open for discussion and feedback to include it.

Following are current change list

1. Update the `Quickstart` guide sample to place `NginxAddOn` after `AwsLoadBalancerControllerAddOn` to avoid `Assertion failed: Missing a dependency for AwsLoadBalancerControllerAddOn` error. As Nginix addon in this case depend on AwsLoadbalancerController to present, so cannot be declare afterward
2. Update the `Quickstart` section on Nginix add on to mention explictly that declaration must be after AWSLoadBalancrController
3. Update the `Quickstart` guide default sample to cover `new XrayAddOn()` to reflect the description that sample will indeed deploy
4. Update the `Quickstart` guide default sample code to cover `ClusterAutoScalerAddOn` to be consistent with description about what is going to deploy
5. Update the `README` guide to remove `team` and `scope` sample (also update Quickstart sample) to avoid compilations error due to undeclare `team` variable

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kcoleman731 — APPROVED** [review]

File: docs/addons/nginx.md [hunk 1]
```diff
@@ -4,16 +4,20 @@ This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ing
 
 Other than handling Kubernetes ingress objects, this ingress controller can facilitate multi-tenancy and segregation of workload ingresses based on host name (host-based routing) and/or URL Path (path based routing). 
 
-This add-on depends on [AWS Load Balancer Controller](aws-load-balancer-controller.md) in order to enable NLB support. 
+***IMPORTANT***: 
+This add-on depends on [AWS Load Balancer Controller](aws-load-balancer-controller.md) in order to enable NLB support.
+
+***AWS Load Balancer Controller Add On must present in Add On array*** and ***must be in addon array before the NGINX ingress controller*** for it to work, as shown in below example. Otherwise will run into error `Assertion failed: Missing a dependency for AwsLoadBalancerControllerAddOn`.
```
[the remark this answers is not in the document]
Comment (danilko) [reply docs/addons/nginx.md:10]: Thanks for pointing out. Did come up with some possible fixes to ensure order does not matter, but seem not 100% ideal. So decided to first push this doc update for the issue to be documented (to help new comers and others).

Now submit an issue about long term fix to address this (analyze + suggested fix) so can discuss further
https://github.com/aws-quickstart/ssp-amazon-eks/issues/225

Thanks again

File: README.md [hunk 2]
```diff
@@ -87,17 +88,20 @@ const addOns: Array<ssp.ClusterAddOn> = [
...
     new ssp.addons.VpcCniAddOn(),
     new ssp.addons.CoreDnsAddOn(),
-    new ssp.addons.KubeProxyAddOn()
+    new ssp.addons.KubeProxyAddOn(),
+    new ssp.addons.XrayAddOn()
 ];
 
-const account = '<YOUR_ACCOUNT_ID'
-const region = 'us-east-2'
+const account = 'XXXXXXXXXXXXX'
```
Comment (shapirov103) [inline README.md:101]: Why this change?

File: README.md [hunk 3]
```diff
@@ -87,17 +88,20 @@ const addOns: Array<ssp.ClusterAddOn> = [
...
+const region = 'us-east-1'
```
Comment (shapirov103) [inline README.md:102]: Why change the region?

File: README.md [hunk 4]
```diff
@@ -87,17 +88,20 @@ const addOns: Array<ssp.ClusterAddOn> = [
...
 const props = { env: { account, region } }
-new ssp.EksBlueprint(scope, { id: 'blueprint', addOns, teams }, props)
+new ssp.EksBlueprint(app, { id: 'blueprint', addOns}, props)
```
Comment (shapirov103) [inline README.md:104]: Nice catch!

File: README.md [hunk 5]
```diff
@@ -113,6 +117,8 @@ If there are no errors you should see the following
 > tsc
 ```
 
+NOTE: If the above setting contains different account/region than initial `cdk bootstrap`, will then need to perfrorm `cdk bootstrap` again to avoid error. Please reference [CDK](https://docs.aws.amazon.com/cdk/latest/guide/home.html) usage doc for detail.
```
Comment (shapirov103) [inline README.md:120]: Note: if the account/region combination used in the code example above is different from the initial combination used with `cdk bootstrap`, you will need to perform `cdk bootstrap` again to avoid errors. 
Please reference [CDK](https://docs.aws.amazon.com/cdk/latest/guide/home.html) usage doc for details.

File: docs/addons/nginx.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shapirov103) [reply docs/addons/nginx.md:10]: Add On = Add-on

File: docs/addons/nginx.md [hunk 6]
```diff
@@ -4,16 +4,20 @@ This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ing
...
 
 ## Usage
 
 ```typescript
-import { NginxAddOn, ClusterAddOn, EksBlueprint }  from '@aws-quickstart/ssp-amazon-eks';
+import { AwsLoadBalancerControllerAddon, NginxAddOn, ClusterAddOn, EksBlueprint }  from '@aws-quickstart/ssp-amazon-eks';
```
Comment (shapirov103) [inline docs/addons/nginx.md:15]: It must be `AwsLoadBalancerControllerAddOn` (note: AddOn vs Addon)

File: docs/addons/nginx.md [hunk 7]
```diff
@@ -4,16 +4,20 @@ This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ing
...
 
 const externalDnsHostname  = ...;
+const awsLBControllerAddOn = new AwsLoadBalancerControllerAddon();
```
Comment (shapirov103) [inline docs/addons/nginx.md:18]: `awsLbControllerAddOn` - camel case always. See `externalDnsHostName` above vs `externalDNSHostName`

File: docs/getting-started.md [hunk 8]
```diff
@@ -40,25 +40,23 @@ import * as ssp from '@aws-quickstart/ssp-amazon-eks';
...
 ];
 
-const opts = { id: 'east-test-1', addOns }
-new ssp.EksBlueprint(app, opts, {
-    env: {
-        account: 'XXXXXXXXXXXXX',
-        region: 'us-east-1'
-    },
-});
+const account = 'XXXXXXXXXXXXX'
```
Comment (shapirov103) [inline docs/getting-started.md:56]: minor: `;` semi-colon after each line

File: README.md (hunk 3 above)
Comment (danilko) [reply README.md:102]: It is actually difference between README and QUICKSTART guide's sample. So thought to be consistent at both place with same code to make future maintenance easier. Is there specific region to consider?

File: docs/addons/nginx.md [hunk 9]
```diff
@@ -4,16 +4,20 @@ This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ing
...
 const addOn = new NginxAddOn({ externalDnsHostname });
```
Comment (askulkarni2) [inline docs/addons/nginx.md:19]: Change this from `addOn` to `nginxAddon` to be similar to `awsLBControllerAddOn`?

File: docs/getting-started.md [hunk 10]
```diff
@@ -69,6 +67,14 @@ Each combination of target account and region must be bootstrapped prior to depl
 cdk bootstrap
 ```
 
+Run the following command to confirm there are no issues with your code
+
+```sh
+npm run build 
+```
+
```
Comment (askulkarni2) [inline docs/getting-started.md:75]: This should not be necessary.

File: README.md (hunk 2 above)
Comment (shapirov103) [reply README.md:101]: No, I suspected consistency, wanted to understand the baseline.

File: docs/addons/nginx.md (hunk 6 above)
Comment (danilko) [reply docs/addons/nginx.md:15]: Update to reflect this change

File: docs/addons/nginx.md (hunk 7 above)
Comment (danilko) [reply docs/addons/nginx.md:18]: Update to reflect the change

File: README.md (hunk 4 above)
Comment (danilko) [reply README.md]: Thanks

File: docs/getting-started.md (hunk 8 above)
Comment (danilko) [reply docs/getting-started.md]: resolved

File: docs/getting-started.md (hunk 10 above)
Comment (danilko) [reply docs/getting-started.md]: resolved.

Update to remove above section about `npm run build` from both `README.md` and `getting-started.md` after discussion

File: docs/addons/nginx.md [hunk 11]
```diff
@@ -4,22 +4,26 @@ This add-on installs [NGINX Ingress Controller](https://kubernetes.github.io/ing
 
 Other than handling Kubernetes ingress objects, this ingress controller can facilitate multi-tenancy and segregation of workload ingresses based on host name (host-based routing) and/or URL Path (path based routing). 
 
-This add-on depends on [AWS Load Balancer Controller](aws-load-balancer-controller.md) in order to enable NLB support. 
+***IMPORTANT***: 
+This add-on depends on [AWS Load Balancer Controller](aws-load-balancer-controller.md) Add-on in order to enable NLB support.
+
+***AWS Load Balancer Controller Add-on must present in Add-on array*** and ***must be in Add-on array before the NGINX ingress controller*** for it to work, as shown in below example. Otherwise will run into error `Assertion failed: Missing a dependency for AwsLoadBalancerControllerAddOn`.
```
Comment (shapirov103) [inline docs/addons/nginx.md:10]: AWS Load Balancer Controller *add-on* must *be* present in *add-on* array

**Review by askulkarni2 — APPROVED** [review]
LGTM!
